internal/controllers/machinedeployment/machinedeployment_sync_test.go

```diff
@@ -495,6 +499,224 @@ func TestSyncDeploymentStatus(t *testing.T) {
+		},
+		Spec: clusterv1.MachineDeploymentSpec{
+			ClusterName:     "test-cluster",
+			Replicas:        pointer.Int32(3),
+			MinReadySeconds: pointer.Int32(10),
+			Strategy: &clusterv1.MachineDeploymentStrategy{
+				Type: clusterv1.RollingUpdateMachineDeploymentStrategyType,
+				RollingUpdate: &clusterv1.MachineRollingUpdateDeployment{
+					MaxSurge:       intOrStrPtr(1),
+					DeletePolicy:   pointer.String("Random"),
+					MaxUnavailable: intOrStrPtr(0),
+				},
+			},
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{"k1": "v1"},
+			},
+			Template: clusterv1.MachineTemplateSpec{
+				ObjectMeta: clusterv1.ObjectMeta{
+					Labels:      map[string]string{"machine-label1": "machine-value1"},
+					Annotations: map[string]string{"machine-annotation1": "machine-value1"},
+				},
+				Spec: clusterv1.MachineSpec{
+					Version:           pointer.String("v1.25.3"),
+					InfrastructureRef: infraRef,
+					Bootstrap: clusterv1.Bootstrap{
+						ConfigRef: &bootstrapRef,
+					},
+					NodeDrainTimeout:        duration10s,
+					NodeVolumeDetachTimeout: duration10s,
+					NodeDeletionTimeout:     duration10s,
+				},
+			},
+		},
+	}
+
+	log := klogr.New()
+
+	skeletonMSBasedOnMD := &clusterv1.MachineSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   "default",
+			Labels:      map[string]string{"machine-label1": "machine-value1"},
+			Annotations: map[string]string{"top-level-annotation": "top-level-annotation-value"},
+		},
+		Spec: clusterv1.MachineSetSpec{
+			ClusterName:     "test-cluster",
+			Replicas:        pointer.Int32(3),
+			MinReadySeconds: 10,
+			DeletePolicy:    string(clusterv1.RandomMachineSetDeletePolicy),
+			Selector:        metav1.LabelSelector{MatchLabels: map[string]string{"k1": "v1"}},
+			Template:        *deployment.Spec.Template.DeepCopy(),
+		},
+	}
+
+	t.Run("should compute a new MachineSet when no old MachineSets exist", func(t *testing.T) {
+		expectedMS := skeletonMSBasedOnMD.DeepCopy()
+
+		g := NewWithT(t)
+		actualMS, err := (&Reconciler{}).computeDesiredMachineSet(deployment, nil, nil, log)
+		g.Expect(err).To(BeNil())
+		assertMachineSet(g, actualMS, expectedMS)
+	})
+
+	t.Run("should compute a new MachineSet when old MachineSets exist", func(t *testing.T) {
+		oldMS := skeletonMSBasedOnMD.DeepCopy()
+		oldMS.Spec.Replicas = pointer.Int32(2)
+
+		expectedMS := skeletonMSBasedOnMD.DeepCopy()
+		expectedMS.Spec.Replicas = pointer.Int32(2) // 4 (maxsurge+replicas) - 2 (replicas of old ms) = 2
+
+		g := NewWithT(t)
+		actualMS, err := (&Reconciler{}).computeDesiredMachineSet(deployment, nil, []*clusterv1.MachineSet{oldMS}, log)
+		g.Expect(err).To(BeNil())
+		assertMachineSet(g, actualMS, expectedMS)
+	})
+
+	t.Run("should compute the updated MachineSet when no old MachineSets exists", func(t *testing.T) {
+		uniqueID := apirand.String(5)
+		existingMS := skeletonMSBasedOnMD.DeepCopy()
+		existingMSUID := types.UID("abc-123-uid")
+		existingMS.UID = existingMSUID
```

Comment:
Can we add a godoc comment here that we expect computeDesiredMachineSet to preserve UID, Name and the unique label and everything else on existingMS to be overwritten?